Add GitHub Actions CI, update version matrix

ibrahima opened the pull request:
Started a simple GitHub Actions workflow from the [Ruby starter workflow](https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml). It doesn't look like it's set up to run yet, not sure if that's because this is a PR and not merged into master.

For now I did a different matrix than what the Travis config had since there are different Rails/Ruby compatibility issues, so not all Ruby versions support all versions of Rails. I ended up going with currently supported versions of Ruby and Rails. I guess I could also have just enumerated the individual combinations like the Travis config had, but I figured if it's not a pricing concern then it might be nice to have a full matrix of tests. Happy to take that in whatever direction makes sense. Thanks!

ibrahima committed:
Add GitHub Actions CI

ibrahima committed:
Add Rails versions to matrix

camertron commented on .github/workflows/ruby.yml:
```diff
@@ -0,0 +1,40 @@
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
```
```suggestion
name: Test
```

camertron commented on .github/workflows/ruby.yml:
```diff
@@ -0,0 +1,40 @@
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
```suggestion
    branches: "*"
```

camertron replied:
As you suspected, I think we're not seeing any jobs run because the workflow is configured to only run for the master branch.

ibrahima replied:
I think this field is actually for which branch the PR is targeting to merge into, so this should have been fine. But I still think you'd probably want to run tests on a PR merging into a different branch anyway.

camertron reviewed (changes requested):
Awesome, thank you! Can we rename ruby.yml -> test.yml?

No problem re: cost, I have quite a few minutes available on my account, and these tests should be pretty fast ⚡

ibrahima committed:
Run on all branches

Co-authored-by: Cameron Dutro <camertron@gmail.com>

ibrahima committed:
Update name

Co-authored-by: Cameron Dutro <camertron@gmail.com>

ibrahima committed:
Rename ruby.yml to test.yml

ibrahima commented:
Ah, you might need to approve the actions run since it's coming from a fork and I am a new contributor. Do you see anything like this? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks (found from [?](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull-request-events-for-forked-repositories))? I checked the settings for my fork and it seems like GitHub Actions are enabled for the fork, and apparently they are actually enabled by default for public repos already.

Thanks! Sorry to bother you with this!

camertron commented:
Hmm unfortunately I don't see any approval button. Here's what I see:

<img width="930" alt="image" src="https://user-images.githubusercontent.com/575280/221263317-89363e7f-f84b-40f1-b664-032ee1c81083.png">

Responding to your earlier comment:

> I think this field is actually for which branch the PR is targeting to merge into, so this should have been fine. But I still think you'd probably want to run tests on a PR merging into a different branch anyway.

I don't think so. This field controls which sorts of events trigger the workflow. Currently it's configured to only run the workflow for the master branch on push or pull request. See the [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters) for more info.

ibrahima commented:
Hmm, did you happen to already check the settings for this repo to see if GitHub Actions is enabled? It sounds like it should be, but perhaps if the repo is old enough it might not be enabled by default? I tested this PR by making a PR into my own fork's master and the actions do seem to run: https://github.com/ibrahima/generated-assets/pull/1. Though it doesn't run successfully, I guess it's not happy about how the Rails version is parametrized currently... will poke at it a little more, maybe add some older Ruby and Rails versions for comparison, maybe it's an issue with specific bundler versions.

ibrahima committed:
Add matrix entries for legacy Rails versions

ibrahima committed:
Set RAILS_VERSION when bundling

ibrahima committed:
Tweak default comment

ibrahima committed:
Delete travis.yml

This wasn't running anyway. I figured adding GH Actions and removing Travis in a single PR would make the transition clearer since both files will show up in the PR.

ibrahima commented:
@camertron I managed to get the Actions working on my fork, see https://github.com/ibrahima/generated-assets/pull/1! I kept the current version list from travis.yml and added a matrix for newer Ruby/Rails versions. Looks like all is still green, so that's good.

ibrahima commented:
> Responding to your earlier comment:
> 
> > I think this field is actually for which branch the PR is targeting to merge into, so this should have been fine. But I still think you'd probably want to run tests on a PR merging into a different branch anyway.
> 
> I don't think so. This field controls which sorts of events trigger the workflow. Currently it's configured to only run the workflow for the master branch on push or pull request. See the [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters) for more info.

You might be right; I updated it anyway just to be safe (even with my interpretation, it makes sense to run specs on branches merging into non-master branches). But I'm thinking it's for the target branch based on the docs for the [pull_request event specifically](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore):

> When using the pull_request and pull_request_target events, you can configure a workflow to run only for pull requests that target specific branches.

I interpreted that to mean that the "branch" field there is for the PR's target branch rather than for the branch being pushed to. The "push" event is still filtered on the "master" branch, since that would trigger for merges into master, though an asterisk there would probably be fine too.

camertron reviewed (approved):
Awesome, this looks great! I still don't see the "Approve and run" button, but let's figure that out post-merge.

camertron commented:
🎉 🎉 Looks like everything ran as expected after merging: https://github.com/camertron/generated-assets/actions/runs/4308576147/jobs/7515010696

Still not sure why it didn't work for this PR/branch, but 🤷 I guess. Hopefully it works for the next one!
